kuma/core/csrf.py

```diff
@@ -0,0 +1,373 @@
+# TODO: Remove this file with Django 1.11
+"""
+Cross Site Request Forgery Middleware.
+
+This module provides a middleware that implements protection
+against request forgeries from other sites.
+
+NOTE: Copied from Django 1.11.12's django.middleware.csrf and slightly
+      modified to work within the context of Django 1.8.19.
+"""
+from __future__ import unicode_literals
+
+import logging
+import re
+import string
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.urlresolvers import get_callable
+from django.utils.cache import patch_vary_headers
+from django.utils.crypto import constant_time_compare, get_random_string
+from django.utils.encoding import force_text
+from django.utils.six.moves import zip
+from django.utils.six.moves.urllib.parse import urlparse
+
+logger = logging.getLogger('kuma.core.csrf')
+
+REASON_NO_REFERER = "Referer checking failed - no Referer."
```

Comment:
This should be under `ugettext` so it can be localizable!

Reply:
These text strings are not localized in Django, and we shouldn't localize them here either. They are sent to the logs, which should be in English regardless of the requester's language, and they shouldn't be printed in the template, which could leak security information.

Reply:
Oh! Sorry, my bad!